api/rpc/handler.go

```diff
@@ -0,0 +1,40 @@
+package rpc
+
+import (
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/das"
+	"github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/share"
+	"github.com/celestiaorg/celestia-node/nodebuilder/state"
+)
+
+var log = logging.Logger("rpc")
+
+type Handler struct {
+	state  state.Module
+	share  share.Module
+	header header.Module
+	das    *das.DASer
+}
+
+func NewHandler(
+	state state.Module,
+	share share.Module,
+	header header.Module,
+	das *das.DASer,
+) *Handler {
+	return &Handler{
+		state:  state,
+		share:  share,
+		header: header,
+		das:    das,
+	}
+}
+
+func (h *Handler) RegisterEndpoints(rpc *Server) {
+	rpc.handler.Register("State", h.state)
```

Comment:
Does namespace have to be capitalised?

Reply:
no, but I did find out that they need to all be the same in the next PR - its only one connection per namespace